### ISSUE TITLE
Closing The Macro Editor

### ISSUE THREAD

**Issue by IHateYourCode** [texstudio-org/texstudio#1825]
### Describe the feature and the current behavior/state

In the `Edit Macros` dialog (`Macros > Edit Macros`) one can close the dialog
by pressing `Escape`, while this is in general useful (for dialogs), doing so while
editing a macro basically means losing all your changes.

**I would suggest to either**

*Implement a check for unsaved macro changes and if true, confirm* 
*closing without saving those changes via a typical dialog window.*

**or**

*Close the dialog via the same functionality as the `Ok` button does,*
*so basically, saving any changes and then closing the dialog.*

### Who will benefit with this feature?
Anyone who edits macros in the `Edit Macros` window
and doesn't want changes to be lost by accident.

### Other Info
If there is a way to configure this via user settings,
I haven't found it. If that is the case, then my feature
request would simply change to have the described
functionality be the default, as it would deem more
useful to the average user to, by default, save
changes rather than loose them.
